module/zfs/dmu_tx.c

```diff
@@ -794,11 +796,20 @@ dmu_tx_delay(dmu_tx_t *tx, uint64_t dirty)
 	 */
 	ASSERT3U(dirty, <, zfs_dirty_data_max);
 
-	now = gethrtime();
-	min_tx_time = zfs_delay_scale *
-	    (dirty - delay_min_bytes) / (zfs_dirty_data_max - dirty);
-	min_tx_time = MIN(min_tx_time, zfs_delay_max_ns);
-	if (now > tx->tx_start + min_tx_time)
+	smoothing_time = 0;
+	delay_time = 0;
+
+	if (dirty > delay_min_bytes) {
+		delay_time = zfs_delay_scale *
+		    (dirty - delay_min_bytes) / (zfs_dirty_data_max - dirty);
```

Comment:
Should probably use the occasion to `READ_ONCE(zfs_dirty_data_max)` at the top of the function, into a `const` value.
Then make all use sites of that variable in this function use the `const` variable

Reply:
Yes, I can try that.